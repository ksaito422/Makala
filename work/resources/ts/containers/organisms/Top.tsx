import React, { useContext } from 'react';
import { Container, Grid, Paper, Typography } from '@material-ui/core';
import { Button } from '../../components/atoms/Button';
import { AuthContext } from '../../contexts/childContexts/AuthContext';
import { StylesContext } from '../../contexts/childContexts/StylesContext';

type Props = {
  registerOnClick: () => void;
  boardOnClick: () => void;
};

export const Top: React.FC<Props> = (props) => {
  /**
   * { 認証状態の確認 }
   * cssの定義
   */
  const { isAuth } = useContext<any>(AuthContext);
  const { useStyles } = useContext<any>(StylesContext);
  const classes = useStyles();

  return (
    <>
      <Container maxWidth='md'>
        <Paper elevation={2} className={classes.paper}>
          <Grid container spacing={4}>
            <Grid item xs={12}>
              <Typography variant='h4' align='center'>
                Welcome to Makala.
              </Typography>
              <br />
              <Typography variant='subtitle1' align='center'>
                Makalaは、文章構成を考案・構築するためのサービスです。
                <br />
                いつ・どこにいても、ブログや記事の内容をささっと書き残すことができます。
              </Typography>
            </Grid>
            <Grid item xs={12} className={classes.centerPlacement}>
              {!isAuth ? (
                // ゲストユーザーなら表示
                <>
                  <Button size='large' onClick={props.registerOnClick}>
                    いますぐはじめる
                  </Button>
                </>
              ) : (
                // ログインユーザーなら表示
                <>
                  <Grid item xs={12}>
                    <Button size='large' onClick={props.boardOnClick}>
                      ボードリストへ
                    </Button>
                  </Grid>
                </>
              )}
            </Grid>
          </Grid>
        </Paper>
      </Container>
    </>
  );
};
