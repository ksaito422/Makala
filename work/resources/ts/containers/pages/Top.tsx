import React from 'react';
import { Header } from '../organisms/Header';
import { Tutorial } from '../organisms/Tutorial';
import {
  Container,
  CssBaseline,
  Grid,
  // useMediaQuery,
} from '@material-ui/core';

export const Top: React.FC = () => {
  return (
    <>
      <CssBaseline />
      <Header
        title='makala'
      />
      <Tutorial />
    </>
  )
}