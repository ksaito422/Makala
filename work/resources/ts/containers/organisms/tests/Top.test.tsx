import React from 'react';
import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import { Top } from '../Top';

// メモリリークを避けるために、各テスト後にすべてのコンテンツを消去する
afterEach(cleanup);

describe('Top', () => {
  it('PCサイズスナップショットテスト', () => {
    window.matchMedia = jest.fn().mockImplementation((query) => {
      return {
        matches: true,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
      };
    });

    const props = {
      registerOnClick: jest.fn(),
      loginOnClick: jest.fn(),
      guestOnClick: jest.fn(),
      boardOnClick: jest.fn(),
    };

    const { baseElement } = render(<Top {...props} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('スマホサイズスナップショットテスト', () => {
    window.matchMedia = jest.fn().mockImplementation((query) => {
      return {
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
      };
    });

    const props = {
      registerOnClick: jest.fn(),
      loginOnClick: jest.fn(),
      guestOnClick: jest.fn(),
      boardOnClick: jest.fn(),
    };

    const { baseElement } = render(<Top {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
