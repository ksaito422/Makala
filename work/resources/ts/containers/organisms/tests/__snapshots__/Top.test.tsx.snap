// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top PCサイズスナップショットテスト 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd"
    >
      <div
        class="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          aria-disabled="false"
          aria-expanded="true"
          class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content Mui-expanded"
          />
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-container MuiCollapse-entered"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <div
                        class="MuiContainer-root MuiContainer-maxWidthMd"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-10"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
                          >
                            <h4
                              class="MuiTypography-root MuiTypography-h4"
                            >
                              Welcome to Makala.
                            </h4>
                            <br />
                            <h6
                              class="MuiTypography-root MuiTypography-subtitle1"
                            >
                              Makalaは、文章構成を考案・構築するためのサービスです。
                              <br />
                              いつ・どこにいても、ブログや記事の内容をささっと書き残すことができます。
                            </h6>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                          >
                            <div
                              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
                            >
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                              >
                                <button
                                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
                                  tabindex="0"
                                  type="button"
                                >
                                  <span
                                    class="MuiButton-label"
                                  >
                                    新規登録する
                                  </span>
                                  <span
                                    class="MuiTouchRipple-root"
                                  />
                                </button>
                              </div>
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                              >
                                <button
                                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
                                  tabindex="0"
                                  type="button"
                                >
                                  <span
                                    class="MuiButton-label"
                                  >
                                    ログインする
                                  </span>
                                  <span
                                    class="MuiTouchRipple-root"
                                  />
                                </button>
                              </div>
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                              >
                                <button
                                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
                                  tabindex="0"
                                  type="button"
                                >
                                  <span
                                    class="MuiButton-label"
                                  >
                                    ゲスト利用する
                                  </span>
                                  <span
                                    class="MuiTouchRipple-root"
                                  />
                                </button>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Top スマホサイズスナップショットテスト 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd"
    >
      <div
        class="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          aria-disabled="false"
          aria-expanded="true"
          class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content Mui-expanded"
          />
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-container MuiCollapse-entered"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <div
                        class="MuiContainer-root MuiContainer-maxWidthXs"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-10"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                          >
                            <h4
                              class="MuiTypography-root MuiTypography-h4"
                            >
                              Welcome to Makala.
                            </h4>
                            <br />
                            <h6
                              class="MuiTypography-root MuiTypography-subtitle1"
                            >
                              Makalaは、文章構成を考案・構築するためのサービスです。
                              <br />
                              いつ・どこにいても、ブログや記事の内容をささっと書き残すことができます。
                            </h6>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                          >
                            <div
                              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
                            >
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                              >
                                <button
                                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
                                  tabindex="0"
                                  type="button"
                                >
                                  <span
                                    class="MuiButton-label"
                                  >
                                    新規登録する
                                  </span>
                                  <span
                                    class="MuiTouchRipple-root"
                                  />
                                </button>
                              </div>
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                              >
                                <button
                                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
                                  tabindex="0"
                                  type="button"
                                >
                                  <span
                                    class="MuiButton-label"
                                  >
                                    ログインする
                                  </span>
                                  <span
                                    class="MuiTouchRipple-root"
                                  />
                                </button>
                              </div>
                              <div
                                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                              >
                                <button
                                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
                                  tabindex="0"
                                  type="button"
                                >
                                  <span
                                    class="MuiButton-label"
                                  >
                                    ゲスト利用する
                                  </span>
                                  <span
                                    class="MuiTouchRipple-root"
                                  />
                                </button>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
